src/Parser/Core/Modules/SpellUsable.js

```diff
@@ -118,6 +118,7 @@ class SpellUsable extends Analyzer {
       cooldown.chargesOnCooldown -= 1;
       this._triggerEvent('updatespellusable', this._makeEvent(spellId, timestamp, 'restorecharge', cooldown));
       this.refreshCooldown(spellId, timestamp);
+      this.reduceCooldown(spellId,remainingCDR,timestamp);
```

Comment:
Please add an if statement so `reduceCooldown` is only called here when `remainingCDR` is not 0. Calling `reduceCooldown` with reduction equal to 0 after almost every `endCooldown` is superfluous, and it might also cause confusing debug logging.